sound/soc/intel/boards/bytcr_rt5640.c

```diff
@@ -1400,10 +1397,16 @@ static int snd_byt_rt5640_mc_probe(struct platform_device *pdev)
 		byt_rt5640_quirk = quirk_override;
 	}
 
+	codec_dev = bus_find_device_by_name(&i2c_bus_type, NULL, byt_rt5640_codec_name);
+	if (!codec_dev)
+		return -EPROBE_DEFER;
+
 	/* Must be called before register_card, also see declaration comment. */
-	ret_val = byt_rt5640_add_codec_device_props(byt_rt5640_codec_name);
-	if (ret_val)
+	ret_val = byt_rt5640_add_codec_device_props(codec_dev, priv);
```

Comment:
Side note. Why this code still present here? AFAICS the only case when the byt_rt5640_codec_name is not NULL is when adev is not NULL, i.e you may reuse adev and add `} else { return -EPROBE_DEFER; }` to the `if (adev) { ... }`.

Reply:
sorry @andy-shev I wasn't able to follow your comment. adev is "struct acpi_device *adev;" , are you suggesting to  add properties to &adev->dev or something?

Reply:
ACPI has a call to get a physical device out of given ACPI device: https://elixir.bootlin.com/linux/latest/source/drivers/acpi/bus.c#L612.

Reply:
thanks @andy-shev for guiding me through the ACPI APIs :-)

Reply:
@plbossart You're welcome! Anyway, this is a side note for possible future improvements, not a blocker for this submission.